### PR TITLE
rebrand and rebase based on firefox-nightly

### PR DESCRIPTION
**Pull request by nyabinary**
second attempt of https://github.com/chaotic-aur/pkgbuild-firefox-wayland-hg/issues/1

### CODE REVIEW

**Review by PedroHLC — DISMISSED** [review]

**Comment by PedroHLC** [comment]
Actually, let me try building it first...

File: PKGBUILD [hunk 1]
```diff
@@ -71,88 +148,119 @@ prepare() {
...
 ac_add_options --enable-application=browser
+mk_add_options MOZ_OBJDIR=${PWD@Q}/obj
 
 ac_add_options --prefix=/usr
 ac_add_options --enable-release
 ac_add_options --enable-hardening
 ac_add_options --enable-optimize
 ac_add_options --enable-rust-simd
-ac_add_options --enable-linker=lld
+ac_add_options --enable-linker=mold
```
Comment (PedroHLC) [inline PKGBUILD]:
```console
 0:01.70 checking for linker...
 0:01.70 ERROR: Unsupported linker mold
```

Looks like mold is not supported without patches.

**Review by PedroHLC — CHANGES_REQUESTED** [review]

File: PKGBUILD (hunk 1 above)
Comment (nyabinary) [reply PKGBUILD]: Fixed

File: .SRCINFO [hunk 2]
```diff
@@ -1,68 +1,74 @@
...
-	makedepends = clang
-	makedepends = llvm
 	makedepends = jack
-	makedepends = nodejs
-	makedepends = cbindgen
-	makedepends = nasm
-	makedepends = python-setuptools
 	makedepends = lld
-	makedepends = dump_syms
+	makedepends = mold
```
Comment (PedroHLC) [inline .SRCINFO]:
```suggestion
```

**Review by PedroHLC — CHANGES_REQUESTED** [review]

File: .SRCINFO (hunk 2 above)
Comment (nyabinary) [reply .SRCINFO]: Hmm, you want to remove the makedepend for mold? I kept it in there just in case in the future we can replace lld for mold.

Comment (PedroHLC) [reply .SRCINFO]: In the future, we can re-add it here, makedepends are never attached as explicit-installed deps anyway!

Comment (nyabinary) [reply .SRCINFO]: Alright going to do that right now

Comment (nyabinary) [reply .SRCINFO]: Done

**Comment by PedroHLC** [comment]
Just for history (since we talked in private):
- There were some MPKG things, leftovers from firefox-nightly, I asked Nyabinary to remove it;
- I asked them to re-add debug symbols and the upload script since Mozilla re-uses this for enriching crash reports.

**Review by PedroHLC — APPROVED** [review]

**Pull request merged**